r/src/array_from_vector.cpp

```diff
@@ -948,6 +974,15 @@ std::shared_ptr<arrow::Array> Array__from_vector(
     Rcpp::stop("Object incompatible with dictionary type");
   }
 
+  // struct types
+  if (type->id() == Type::STRUCT) {
+    if (type_infered || arrow::r::CheckCompatibleStruct(x, type)) {
```

Comment:
I'd make `type_infered` mandatory (`&&`) until CheckCompatibleStruct is well defined. Otherwise you might wander in segfault-land.

Reply:
sure, but it was the very next thing on the list, done now.